internal/common/objects/MegaBike.go

```diff
@@ -123,21 +120,27 @@ func (mb *MegaBike) GetKickedOutCount() int {
 	return mb.kickedOutCount
 }
 
-func (mb *MegaBike) KickOutAgent() map[uuid.UUID]int {
-	voteCount := make(map[uuid.UUID]int)
+// only called for level 0 and level 1
+func (mb *MegaBike) KickOutAgent(weights map[uuid.UUID]float64) []uuid.UUID {
+	voteCount := make(map[uuid.UUID]float64)
 	// Count votes for each agent
 	for _, agent := range mb.agents {
 		agentVotes := agent.VoteForKickout() // Assuming this now returns map[uuid.UUID]int
 		for agentID, votes := range agentVotes {
-			voteCount[agentID] += votes
+			agentWeight := weights[agentID]
+			if val, ok := voteCount[agentID]; ok {
```

Comment:
Not sure if VoteCount[] has a value at this point as it seems like it would be empty at this point